neuro_extras/main.py

```diff
@@ -417,11 +321,55 @@ async def _data_cp(
                     # gunzip removes src after extraction, while tar - not
                     file.unlink()
 
+        if compress:
+            file = Path(destination_url.path.split("/")[-1])
+            tmp_dst_archive = Path(TEMP_UNPACK_DIR / file)
+            suffixes = file.suffixes
+            if suffixes[-2:] == [".tar", ".gz"] or suffixes[-1] == ".tgz":
+                command = "tar"
+                args = ["zcf", str(tmp_dst_archive), "-C", str(source_url.path), "."]
+            elif suffixes[-2:] == [".tar", ".bz2"] or suffixes[-1] in (".tbz2", ".tbz"):
+                command = "tar"
+                args = ["jcf", str(tmp_dst_archive), str(source_url.path)]
+            elif suffixes[-1] == ".tar":
+                command = "tar"
+                args = ["cf", str(tmp_dst_archive), str(source_url.path)]
+            elif suffixes[-1] == ".gz":
+                command = "gzip"
+                args = [str(tmp_dst_archive), str(source_url.path)]
+            elif suffixes[-1] == ".zip":
+                command = "zip"
+                args = [str(tmp_dst_archive), str(source_url.path)]
```

Comment:
Do we support directories as sources?

Reply:
Yes, we do. Why do you ask? Does this code look wrong in that regard?

Reply:
Hm, I would say that we should add `-r` param for `zip`, and maybe raize an error for bare `gzip`, since it cannot compress directories, only files.

Reply:
Good catch for `zip`, added!
As far as `gzip` is concerned, I would imagine it's a super rare case to create a gzip archive in general, and from directories in particular. Maybe detection could isn't worth the effort as `gzip` will fail with a corresponding error message anyway. What do you think?

Reply:
Sounds reasonable to me. @mariyadavydova, @yartem, what do you think?

Reply:
Sounds reasonable for me as well.